plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2754,9 +2911,16 @@ private List<CommitInfoEntry> getCommitInfoEntries(SchemaTableName table, String
 
     private static ColumnMetadata getColumnMetadata(DeltaLakeColumnHandle column, @Nullable String comment, boolean nullability, @Nullable String generation)
     {
+        String columnName = column.getBaseColumnName();
+        Type columnType = column.getBaseType();
+        if (!column.isBaseColumn()) {
+            DeltaLakeColumnProjectionInfo projectionInfo = column.getProjectionInfo().get();
+            columnName = column.getQualifiedPhysicalName();
+            columnType = projectionInfo.getType();
+        }
```

Comment:
avoid assigning before you know information is true

```suggestion
        String columnName = column.getBaseColumnName();
        Type columnType = column.getBaseType();
        if (column.isBaseColumn()) {
           columnName = column.getBaseColumnName();
           columnType = column.getBaseType();
           } 
           else {
            DeltaLakeColumnProjectionInfo projectionInfo = column.getProjectionInfo().get();
            columnName = column.getQualifiedPhysicalName();
            columnType = projectionInfo.getType();
        } 
```